src/set_theory/game/rgame.lean

```diff
@@ -0,0 +1,153 @@
+/-
+Copyright (c) 2022 Violeta Hernández Palacios. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Violeta Hernández Palacios
+-/
+
+import set_theory.game.pgame
+
+/-!
+# Pre-games up to relabelling
+
+In `set_theory/pgame.lean`, we define combinatorial pre-games as arising from a left and right
+family of pre-games. Despite being the most natural definition in type theory, this leads to a major
+inconvenience, where operations on pre-games aren't as well-behaved as they are in ZFC. For
+instance, pre-game addition isn't commutative, since the indexing types for the moves don't match.
+
+Here, we define the quotient `rgame` of `pgame` by relabellings. Two games are relabellings of one
+another if there exists a bijection between their left and right moves, and the same is true for
+their left and right options. We prove that `rgame` is an `add_comm_monoid`.
+
+## Todo
+
+- Move the definition of `pgame.mul` here, prove basic results about `rgame.mul`.
+- Define the relation of subsequency up to relabelling.
+-/
+
+open pgame function
+
+universe u
+
+/-- The setoid of pre-games quotiented by the existence of a relabelling. -/
+def rgame.setoid : setoid pgame :=
+⟨λ x y, nonempty (relabelling x y),
+ λ x, ⟨relabelling.refl x⟩,
+ λ x y ⟨h⟩, ⟨h.symm⟩,
+ λ x y z ⟨h₁⟩ ⟨h₂⟩, ⟨h₁.trans h₂⟩⟩
+
+/-- The quotient of pre-games quotiented by the existence of a relabelling. -/
+def rgame := quotient rgame.setoid
+
+namespace rgame
+
+/-- Lift an relabelling-respecting function on `pgame` to `rgame`. -/
+def lift {α} (f : pgame → α)
+  (H : ∀ {x y}, relabelling x y → f x = f y) : rgame → α :=
+quot.lift f (λ x y h, begin
+  change nonempty (relabelling x y) at h,
+  cases h,
+  exact H h
+end)
+
+/-- Lift a binary relabelling-respecting function on `pgame` to `rgame`. -/
+def lift₂ {α} (f : pgame → pgame → α)
+  (H : ∀ {x₁ y₁ x₂ y₂}, relabelling x₁ x₂ → relabelling y₁ y₂ → f x₁ y₁ = f x₂ y₂) :
+  rgame → rgame → α :=
+@quotient.lift₂ _ _ _ rgame.setoid rgame.setoid f (λ x₁ y₁ x₂ y₂ ⟨hx⟩ ⟨hy⟩, H hx hy)
+
+/-- Lift a pre-game to its quotient by relabelling. -/
+def mk (x : pgame) : rgame := quotient.mk' x
+
+theorem sound {x y : pgame} (e : relabelling x y) : mk x = mk y :=
+quot.sound ⟨e⟩
+
+@[elab_as_eliminator] theorem induction_on {p : rgame → Prop} (x : rgame) : (∀ a, p (mk a)) → p x :=
+quotient.induction_on' x
+
+instance : has_le rgame :=
+⟨lift₂ (λ x y, x ≤ y) (λ x₁ y₁ x₂ y₂ hx hy, propext (le_congr hx.equiv hy.equiv))⟩
+
+theorem le_rfl : ∀ {x : rgame}, x ≤ x :=
+by { rintro ⟨x⟩, exact pgame.le_rfl }
+theorem le_refl (x : rgame) : x ≤ x :=
+le_rfl
+theorem le_trans : ∀ x y z : rgame, x ≤ y → y ≤ z → x ≤ z :=
```

Comment:
I think these should be protected. Or even just inlined in the `preorder` instance.

Reply:
I would inline them.

Reply:
Having a `preorder` instance right now would be really confusing, given that `<` on `rgame` would have a different definition than `<` on `pgame`. I'll block this until #13963 gets merged.